tests/appsec/iast/utils.py

```diff
@@ -16,18 +16,18 @@ def _get_expectation(d):
     raise TypeError(f"Unsupported expectation type: {d}")
 
 
-def _get_span_meta(request):
+def _get_span_meta(request, metastruct=False):
     spans = [span for _, span in interfaces.library.get_root_spans(request=request)]
     assert spans, "No root span found"
     span = spans[0]
     meta = span.get("meta", {})
-    return meta
+    meta_struct = span.get("meta_struct", {})
+    return meta, meta_struct
 
 
 def get_iast_event(request):
-    meta = _get_span_meta(request=request)
-    assert "_dd.iast.json" in meta, "No _dd.iast.json tag in span"
-    return meta["_dd.iast.json"]
+    meta, meta_struct = _get_span_meta(request=request)
+    return meta.get("_dd.iast.json") or meta_struct.get("iast")
```

Comment:
Could you add an assertion that check the event is in one place ? otherwise, it will fail later on an `None`, which is harder to understand. Something like : 

```
assert "_dd.iast.json" in meta or "iast" in meta_struct, "No IAST info found tag in span"

Reply:
Applied in the latest commits 😄